File: Data Structures & Algorithms/reverse-a-linked-list/submission-0.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */

class Solution {
public:
    ListNode* reverseList(ListNode* head) {
        ListNode* prev = nullptr;
        ListNode* current = head;
        while(current != nullptr) {
            ListNode* temp = current->next;
            //cout << "Temp: " << temp->val << endl;
            current->next = prev;
            prev = current;
            current = temp;
            //cout << "Temp: " << temp->val << endl;
        }
        return prev;
    }
};

File: Data Structures & Algorithms/climbing-stairs/submission-0.py
class Solution:
    def climbStairs(self, n: int) -> int:
        if n == 0:
            return 0
        if n == 1:
            return 1
        if n == 2:
            return 2
        current = [0] * (n + 1)
        current[1] = 1
        current[2] = 2
        for i in range(3, n + 1):
            current[i] = current[i - 1] + current[i - 2]
        return current[-1]
